### PR TITLE
docs: fix site FOUC

### PR DESCRIPTION
**Pull request by MadCcc**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
由于 https://github.com/ant-design/ant-design/pull/49550 回滚了 dumi 版本，在 https://github.com/ant-design/ant-design/pull/49109 删掉的 cssinjs 的提取逻辑需要加回去。
升级 dumi ssr 时需要再删掉 @Jinbao1001 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
> cssinjs 的提取逻辑需要加回去

佬，哪个PR删除的关联一下？

**Comment by MadCcc** [comment]
> 佬，哪个PR删除的关联一下？

已贴

File: .dumi/theme/plugin.ts [hunk 1]
```diff
@@ -197,6 +217,30 @@ const RoutesPlugin = (api: IApi) => {
...
+        file.content = addLinkStyle(file.content, cssFile, true);
+
+        // Insert antd cssVar to head
+        const cssVarMatchRegex = /<style data-type="antd-css-var"[\S\s]+?<\/style>/;
+        const cssVarMatchList = file.content.match(cssVarMatchRegex) || [];
+
+        cssVarMatchList.forEach((text) => {
+          file.content = file.content.replace(text, '');
+          file.content = file.content.replace('<head>', `<head>${text}`);
+        });
```
Comment (afc163) [inline .dumi/theme/plugin.ts:242]: 这么复杂，需要封装到 cssinjs 的 extractStyle 里么？

Comment (MadCcc) [reply .dumi/theme/plugin.ts:242]: 不封装，以后不需要了，umi 会自动处理

**Review by zombieJ — APPROVED** [review]

**Pull request merged**